sql-data-pipeline/src/main/java/io/cdap/pipeline/sql/app/core/AbstractSQLExecutor.java

```diff
@@ -0,0 +1,402 @@
+  private List<String> queries;
+  private boolean createTableFeatureFlag;
+
+  public AbstractSQLExecutor(SQLConfig config, Map<String, SQLTransform> pluginMap) {
+    this.config = config;
+    this.sourceStages = new HashSet<>();
+    this.sinkStages = new HashSet<>();
+    this.pluginMap = pluginMap;
+    this.fromNodeMap = new HashMap<>();
+    this.toNodeMap = new HashMap<>();
+    this.relationalNodeMap = new HashMap<>();
+    this.temporaryTableMap = new HashMap<>();
+    this.queries = new ArrayList<>();
+    this.createTableFeatureFlag = false;
+  }
+
+  /**
+   * Returns the list of queries which were generated by the planner. init() must be called prior to
+   * retrieving the queries, otherwise this could return an empty list.
+   * @return The generated query plan
+   */
+  public List<String> getQueries() {
+    return queries;
+  }
+
+  /**
+   * Returns the {@link SqlDialect} to use for the concrete implementation of this class.
+   * @return The dialect for the concrete implementation
+   */
+  public abstract SqlDialect getDialect();
+
+  @Override
+  public void configure() {
+    Gson gson = new GsonBuilder().create();
+
+    // Property map for storing runtime state
+    Map<String, String> properties = new HashMap<>();
+    // Initialize the executor state and store in property map
+    initExecutorState(gson, properties);
+
+    // Load the Calcite JDBC driver
+    loadCalciteJdbcDriver();
+    // Traverse the topological ordering to prepare the queries for validation
+    traverseTopologicalOrder();
+  }
+
+  @Override
+  public void initialize() {
+    // Set the config
+    Gson gson = new GsonBuilder().create();
+
+    // Flag to enable or disable creation of tables when they don't exist
+    createTableFeatureFlag = false;
+    // Set whether we want to create tables if they do not exist
+    if (getContext().getRuntimeArguments().get(CREATE_TABLE_FEATURE_FLAG) == "true") {
+      createTableFeatureFlag = true;
+    }
+
+    // Load the dag and other properties previously initialized at configure time
+    loadExecutorState(gson, getContext().getSpecification().getProperties());
+
+    // Load the Calcite JDBC driver
+    loadCalciteJdbcDriver();
+    // Traversal Logic to prepare the query ordering
+    traverseTopologicalOrder();
+
+    // Serialize all sink and splitter RelNodes
+    // Converter to convert RelNodes to SqlNodes
+    RelToSqlConverter converter = new RelToSqlConverter(getDialect());
+    // Creates a temporary table for each splitter
+    createTemporaryTables(converter);
+    // Creates a query for each sink
+    createSinkQueries(converter);
+    // Make the query list unmodifiable
+    queries = Collections.unmodifiableList(queries);
+  }
+
+  /**
+   * Initializes the state of the executor and stores the GSON of the state in the properties map. This should
+   * only happen at configure time.
+   */
+  private void initExecutorState(Gson gson, Map<String, String> properties) {
```

Comment:
Call this `initExecutor` or `initExecutorConfig`, since I don't see any state in the implementation

Reply:
Fixed